velox/exec/ContainerRowSerde.cpp

```diff
@@ -522,33 +514,29 @@ int compare<TypeKind::ARRAY>(
 }
 
 template <>
-int compare<TypeKind::MAP>(
+std::optional<int32_t> compare<TypeKind::MAP>(
     ByteStream& left,
     const BaseVector& right,
     vector_size_t index,
     CompareFlags flags) {
-  VELOX_DCHECK(!flags.mayStopAtNull(), "not supported null handling mode");
-
   auto map = right.wrappedVector()->asUnchecked<MapVector>();
   VELOX_CHECK_EQ(map->encoding(), VectorEncoding::Simple::MAP);
   auto wrappedIndex = right.wrappedIndex(index);
   auto size = map->sizeAt(wrappedIndex);
   std::vector<vector_size_t> indices(size);
   auto rightIndices = map->sortedKeyIndices(wrappedIndex);
   auto result = compareArrayIndices(left, *map->mapKeys(), rightIndices, flags);
-  if (result) {
+  if (result.has_value() && result.value() != 0) {
```

Comment:
nit: `if (result && *result != 0)` for consistency with other places?

Reply:
Thanks for your remind. I've fixed it using the following code, and improve the ut of MapType.
```C++
if (result && *result == 0) {
    return compareArrayIndices(left, *map->mapValues(), rightIndices, flags);
  }
  return result;
```